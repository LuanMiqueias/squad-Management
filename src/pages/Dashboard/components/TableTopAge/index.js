***REMOVED***
import styles from "./style.module.css";

// import iconDelete from "../../assets/icons/delete.svg";
// import iconShare from "../../assets/icons/share.svg";
// import iconEdit from "../../assets/icons/edit.svg";

const TableTopAge = ({ data, title }) => {
***REMOVED***
    <div className={styles.container}>
      <div>
        <h2 className={styles.title}>{title}</h2>
      </div>
      <div className={styles.table}>
        {data.map(({ name, description }, index) => {
        ***REMOVED***
            <div className={styles.content} key={`${name}_${index}`}>
              <span className={styles.TableItems} aria-label="name">
                {name}
              </span>
              <span className={`${styles.TableItems}`} aria-label="description">
                {description}
              </span>
    ***REMOVED***
  ***REMOVED***
  ***REMOVED***)}
      </div>
    </div>
***REMOVED***
***REMOVED***

export default TableTopAge;
