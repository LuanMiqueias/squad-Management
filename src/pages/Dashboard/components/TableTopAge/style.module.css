.container{
  width: 100%;
}
.title{
  font-size: .88rem;
  color:var(--text);
  font-weight: 600;
  padding: 8px 0;
}
.table{
  padding: 4px;
  font-size: .75rem;
  font-weight: 400;
  color:var(--black);
  background:var(--gray);
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  gap:4px;
}

.content{
  display: flex;
  gap:24px;
  padding: 2px 8px;
  transition: all .05s;
  background: #fff;
  border-radius: 6px;
  justify-content: space-between;
}
.content .TableItems{
  padding: 8px 16px;  
}
.TableItems:nth-child(2){
  font-weight: 600;
}

