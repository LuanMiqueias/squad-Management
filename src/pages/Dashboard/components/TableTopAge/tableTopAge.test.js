import { render } from "@testing-library/react";
import TableTopAge from "./index";

const props = [
  {
    name: "test 1",
    description: "description 1",
***REMOVED***,
  {
    name: "teste 2",
    description: "Description 1",
***REMOVED***,
];

it("Should return name and description", () => {
  const { getAllByLabelText } = render(<TableTopAge data={props} />);
  const names = getAllByLabelText("name");
  const descriptions = getAllByLabelText("description");

  names.forEach((name) => {
    return expect(name).not.toHaveTextContent("");
***REMOVED***);
  descriptions.forEach((description) => {
    return expect(description).not.toHaveTextContent("");
***REMOVED***);
});

it("should match snapshot", () => {
  const { container } = render(<TableTopAge data={props} />);
  expect(container.firstChild).toMatchSnapshot();
});
