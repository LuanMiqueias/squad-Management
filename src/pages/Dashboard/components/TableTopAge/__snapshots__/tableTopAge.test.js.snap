// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="container"
>
  <div>
    <h2
      class="title"
    />
  </div>
  <div
    class="table"
  >
    <div
      class="content"
    >
      <span
        aria-label="name"
        class="TableItems"
      >
        test 1
      </span>
      <span
        aria-label="description"
        class="TableItems"
      >
        description 1
      </span>
    </div>
    <div
      class="content"
    >
      <span
        aria-label="name"
        class="TableItems"
      >
        teste 2
      </span>
      <span
        aria-label="description"
        class="TableItems"
      >
        Description 1
      </span>
    </div>
  </div>
</div>
`;
